presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/functions/PinotDistinctCountAggregationFunction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.pinot.functions;
+
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.spi.function.AccumulatorState;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+import io.airlift.slice.Slice;
+
+@Description("Pinot UDF: A distinct count function, should always be pushed down to Pinot")
+@AggregationFunction("distinctCount")
```

Comment:
@fx19880617 ... can you comment more on what happens if we don't have this PinotDistinctCountAggregationFunction ? We already seem to have a translation for Presto's distinctcount to Pinot's DISTINCTCOUNT, so it's not clear to me why we have this function ?

Reply:
For this one, we are not missing the functionality support for `distinctCount`, but more or less a semantic align with this `segmentPartitionedDistinctCount`(https://github.com/apache/incubator-pinot/pull/5786).

The thing we want to achieve is to allow more functionalities to be pushed down to pinot side to be executed.

This PR is more like a sample PR for UDFs and apart from these two, we will have more functions to be added.

Reply:
Okay, so if this PR is meant to be just an example, do we *really need* the file `PinotDistinctCountAggregationFunction.java` ? Like, I am curious if the sample query 1 in the PR description will now fail if we don't have this file present ?

(We still have the second file for the example purposes)

Reply:
I feel like so, as users can directly copy over the pinot query here and run it in presto.